synapse/federation/federation_base.py

```diff
@@ -67,13 +79,10 @@ async def _check_sigs_and_hash(
         """
         try:
             await _check_sigs_on_pdu(self.keyring, room_version, pdu)
-        except SynapseError as e:
-            logger.warning(
-                "Signature check failed for %s: %s",
-                pdu.event_id,
-                e,
-            )
-            raise
+        except InvalidEventSignatureError as e:
+            errmsg = f"event id {pdu.event_id}: {e}"
+            logger.warning("%s", errmsg)
+            raise SynapseError(403, errmsg, Codes.FORBIDDEN)
```

Comment:
`_check_sigs_on_pdu` now raises an `InvalidEventSignatureError` where previously it raised a `SynapseError`, so modulo slight wording tweaks to the log line and returned error, this should be identical, functionally.